src/Analytics/Adapter.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * Utopia PHP Framework
+ *
+ * @package Analytics
+ * @subpackage Tests
+ *
+ * @link https://github.com/utopia-php/framework
+ * @author Torsten Dittmann <torsten@appwrite.io>
+ * @version 1.0 RC1
+ * @license The MIT License (MIT) <http://www.opensource.org/licenses/mit-license.php>
+ */
+
+namespace Utopia\Analytics;
+
+abstract class Adapter
+{
+    protected $enabled = true;
+    public $endpoint = '';
+
+    const METHOD_GET = 'GET';
+    const METHOD_POST = 'POST';
+    const METHOD_PUT = 'PUT';
+    const METHOD_PATCH = 'PATCH';
+    const METHOD_DELETE = 'DELETE';
+    const METHOD_HEAD = 'HEAD';
+    const METHOD_OPTIONS = 'OPTIONS';
+    const METHOD_CONNECT = 'CONNECT';
+    const METHOD_TRACE = 'TRACE';
+
+    /**
+     * Enables tracking for this instance.
+     * 
+     * @return void
+     */
+    public function enable(): void
+    {
+        $this->enabled = true;
+    }
+
+    /**
+     * Disables tracking for this instance.
+     * 
+     * @return void
+     */
+    public function disable(): void
+    {
+        $this->enabled = false;
+    }
+
+        /**
+     * Call
+     *
+     * Make an API call
+     *
+     * @param string $method
+     * @param string $path
+     * @param array $params
+     * @param array $headers
+     * @return array|string
+     * @throws = Utopia\Analytics\AnalyticsException
+     */
+    public function execute($method, $path = '', $headers = array(), array $params = array())
```

Comment:
Execute should be an abstract method implemented by each adapter. 

`abstract public function execute();`